src/app/reporting/Engine.cpp

```diff
@@ -848,6 +848,10 @@ CHIP_ERROR Engine::SendReport(ReadHandler * apReadHandler, System::PacketBufferH
     // We can only have 1 report in flight for any given read - increment and break out.
     mNumReportsInFlight++;
     err = apReadHandler->SendReportData(std::move(aPayload), aHasMoreChunks);
+    if (err != CHIP_NO_ERROR)
+    {
+        --mNumReportsInFlight;
```

Comment:
Could this ever underrun where it is used elsewhere

Reply:
The way things work with this PR (and it's still not great, but I was trying to minimize changes so we can backport to 1.0) is:
1. If `SendReportData` fails, the ReadHandler guarantees that it's not in the `AwaitingReportResponse` state.
2. The only places we decrement this counter are here and in `OnReportConfirm`
3. `OnReportConfirm` is called in three cases:
    a. `SendReportData` succeeded but we are not waiting for a response (in which case we do not enter `AwaitingReportResponse`).
    b. `ReadHandler` leaves `AwaitingReportResponse` state.
    c. `ReadHandler` in `AwaitingReportResponse` state is destroyed.

So I don't believe this can underflow: The increment in `Engine::SendReport` is exactly balanced by one of those cases.

Reply:
Understood it cannot underflow *now*, but it would be important that it not in the future as well :)

Reply:
Yes, I agree.  In the future, I'd really like to just remove this counter.